src/lib/client.ts

```diff
@@ -852,13 +856,17 @@ export default class MqttClient extends TypedEventEmitter<MqttClientEventCallbac
 			protocolId: this.options.protocolId,
 			protocolVersion: this.options.protocolVersion,
 			clean: this.options.clean,
-			clientId: this.options.clientId,
 			keepalive: this.options.keepalive,
 			username: this.options.username,
-			password: this.options.password as Buffer,
+			// @ts-expect-error - @TODO: @robertsLando this is a dangerous situation as this.options.password can never be a Buffer but was forced as a Buffer
+			password: this.options.password,
```

Comment:
I know, there was a PR on mqtt-packet to fix this type don't remember right now if it has been merged or not but it was safe to force it to buffer especially because almost everyone passes it as string